vm/devices/user_driver/src/vfio.rs

```diff
@@ -100,6 +115,10 @@ impl VfioDevice {
         }
 
         container.set_iommu(IommuType::NoIommu)?;
+        if keepalive {
+            // Prevent physical hardware interaction when restoring.
+            group.set_keep_alive(path.file_name().unwrap().to_str().unwrap())?;
```

Comment:
`path.file_name().unwrap().to_str().unwrap()` is just `pci_id`, right?

Reply:
/sys/bus/pci/devices/<pci_id>

Reply:
yes, file_name() should be just pci_id. Let me test it

Reply:
Tested, working.